Test failure: Master and 6.x failing for since January 24

https://logstash-ci.elastic.co/job/elastic-logstash-6.x/ and https://logstash-ci.elastic.co/job/elastic-logstash-master/

Some of the dependent jobs are succeeding, but at least one has failed every time for the past 15 days.

The slave wa replaced and we still get the same error
```
16:46:39     [ERROR] 2018-02-10 16:46:38.961 [LogStash::Runner] pipeline - Logstash failed to create queue {:pipeline_id=>"test-without-dlq", "exception"=>"Not enough free disk space available to allocate persisted queue.", "backtrace"=>["org.logstash.ackedqueue.Queue.ensureDiskAvailable(Queue.java:854)", "org.logstash.ackedqueue.Queue.open(Queue.java:157)", 
```

The problem here is that I can't repro locally.

I just remembered that one thing that is different is the `subst` trick we do on Windows CI to remap too-long paths to a drive letter. I am wondering if that might be causing problems. Investigating.

--- a/logstash-core/src/main/java/org/logstash/common/FsUtil.java
+++ b/logstash-core/src/main/java/org/logstash/common/FsUtil.java
@@ -5,6 +5,8 @@ import java.io.IOException;
 import java.util.Arrays;
 import java.util.HashSet;
 import java.util.Set;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
 /**
  * File System Utility Methods.
@@ -13,6 +15,9 @@ public final class FsUtil {
 
     private FsUtil() {
     }
+
+    private static final boolean IS_WINDOWS = System.getProperty("os.name").startsWith("Windows");
+    private static final Logger logger = LogManager.getLogger(FsUtil.class);
 
     /**
      * Checks if the request number of bytes of free disk space are available under the given
@@ -34,6 +39,18 @@ public final class FsUtil {
             if (location == null) {
                 throw new IllegalStateException(String.format("Unable to determine the partition that contains '%s'", path));
             }
+        }
+
+        final long freeSpace = location.getFreeSpace();
+
+        if (freeSpace == 0L && IS_WINDOWS) {
+            // On Windows, SUBST'ed drives report 0L from getFreeSpace().
+            // The API doc says "The number of unallocated bytes on the partition or 0L if the abstract pathname does not name a partition."
+            // There is no straightforward fix for this and it seems a fix is included in Java 9.
+            // One alternative is to launch and parse a DIR command and look at the reported free space.
+            // This is a temporary fix to get the CI tests going which relies on SUBST'ed drives to manage long paths.
+            logger.warn("Cannot retrieve free space on " +  location.toString() +  ". This is probably a SUBST'ed drive.");
+            return true;
         }
 
         return location.getFreeSpace() >= size;

